### PR TITLE
common,mds,osd: Explicitly delete copy ctor if noncopyable

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/osdc/Objecter.h [hunk 1]
```diff
@@ -1755,8 +1755,8 @@ class Objecter : public md_config_obs_t, public Dispatcher {
 		 map_dne_bound(0) {}
 
     // no copy!
```
Comment (tchaikov) [inline src/osdc/Objecter.h]: might want to delete the comment of `// no copy!`, as the code is self-documenting.

**Comment by shinobu-x** [comment]
test this please

File: src/mds/Capability.h [hunk 2]
```diff
@@ -127,9 +127,9 @@ class Capability : public Counter<Capability> {
     mseq(0),
     suppress(0), state(0) {
   }
-  Capability(const Capability& other);  // no copying
+  Capability(const Capability& other) = delete;  // no copying
```
Comment (tchaikov) [inline src/mds/Capability.h]: and these two.

Comment (shinobu-x) [reply src/mds/Capability.h]: good catch!

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-12-14_08:53:42-rados-wip-kefu-testing-2017-12-14-1157-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-12-15_06:17:13-rados-wip-kefu-testing-2017-12-14-1157-distro-basic-mira/

**Pull request merged**